compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -2405,6 +2406,7 @@ uint8_t* TR::X86RegMemInstruction::generateOperand(uint8_t* cursor)
 
    if ((getOpCode().isEvexInstruction() && getEncodingMethod() != OMR::X86::Legacy) || getEncodingMethod() >= OMR::X86::EVEX_L128)
       {
+      toRealRegister(getTargetRegister())->setSource2ndRegisterFieldInEVEX(cursor - 4);
```

Comment:
I identified that this logic does not hold if the EVEX only supports 2-operands (such as movdqu).

Reply:
Added an instruction flag to indicate the the opcode is unary, even when EVEX encoded.